### DISCUSSION TITLE
Sunbird Admin Portal & Dashboard (V1.0)

### DISCUSSION THREAD

**Discussion by poojakarma** [Sunbird-Ed/Community#623] [Contribution]
Currently Sunbird does not support  tenant creation, user creation etc.. via ui.  So we have started building a UI for putting together all such tasks to benefit the community. Sharing the [repo](https://github.com/tekdi/sunbird-admin-ui) with the details .

**Name: Sunbird Admin Portal & Dashboard**
Version: 1.0

**Currently Implemented features :**
1. Organisation Dashboard -> Shows Organisations, Number of sub-Organisation, Number of User and Roles/content Information
Display Shot:
![Screenshot from 2023-09-21 19-00-10](https://github.com/Sunbird-Ed/Community/assets/126868748/f05f80f4-72cc-491f-8d33-48e360b8360d)

2. Create and modify Organisation ,create Sub-Organisation
Display Shot: 
![Screenshot from 2023-09-21 19-00-50](https://github.com/Sunbird-Ed/Community/assets/126868748/6c6ddbe6-5750-4a90-a9ea-1c2f7612e5ff)

3. Create User in Organization and sub-Organization, Edit Role of User , Block/unblock user
Display Shot: 
![Screenshot from 2023-10-25 14-55-31](https://github.com/Sunbird-Ed/Community/assets/126868748/45f370df-b94f-44ca-b909-ab24a289a254)

4. User Dashboard -> Shows total users,and User list with filter feature
Display Shot: 
![Screenshot from 2023-09-21 19-01-04](https://github.com/Sunbird-Ed/Community/assets/126868748/ae32d51c-48d2-40c7-918a-8ec596495acf)

**Features in pipeline :**
1. Click and view reports (eg. Course reports)
2. Certificate Re-issue & Share
3. User Bulk Upload
4. Content bulk upload
5. KeyCloak Integration

**Install and Setup**
**1) Prerequisites**
Angular CLI: 15.2.6
Node 14.21.3
Npm 6.14.18
Authkey and User token of the SB instance that you want to connect / perform the admin tasks 

**2) Steps to Install**
Step 1: fork from https://github.com/tekdi/sunbird-admin-ui.git here
Step 2: create new fork  
Step 3: open VS code , open terminal and enter below command
Step 4:  git clone  https://github.com/your_repo_name/sunbird-admin.git (fork Url)
Step 5 : cd [sunbird-admin](https://github.com/your_repo_name/sunbird-admin.git)
Step 6: git remote add upstream https://github.com/tekdi/sunbird-admin-ui.git (main git Url)
Step 7:  git checkout -b “admin-Dashboard”
Step 8:  git fetch upstream main
Step 9:  git pull upstream main
Step 10 : npm i
Step 11: paste below code in environment.ts file and set authKey and userToken
Path : src -> environments -> environment.ts
`export const environment = {
production: false,
port: "http://localhost:4200/",
instance: "https://your-sunbird-instance.com/",
apiUrl: "https://your-sunbird-instance.com/api",
target: "https://your-sunbird-instance.com",
addOrgUrl: "api/org/v1/create",
orgUrl: "api/org/v1/search",
userUrl: "api/user/v1/search",
userCreateUrl: "api/user/v3/create",
userUpdateRoleUrl: "api/user/v1/role/assign",
blockUserUrl: "api/user/v1/block",
unblockUserUrl: "api/user/v1/block",
authKey: "",
userToken: "",
"expires_in": 21600,
"refresh_expires_in": 21600,
"refresh_token": "",
};
`
Step 12: set url in proxy.conf.js file 
`"target": "https://your-sunbird-instance.com/"` 
// The sb instance that you want connect to perform these tasks`
Step 13  Run the command  ng serve from the project directory 
Step 14  Open the browser and type http://localhost/:4200

**3. Important Configuration files**
1. config->constant.config.ts (SystemRoles) : Users can easily extend SystemRoles by adding new system roles into SystemRoles Constant to efficiently control access and visibility within the system.
2. config->constant.config.ts (Roles) : Users can easily add more roles options to create user under particular role by adding in Roles constant

**Comment by Krishnaj20** [comment]
@poojakarma - thank you for sharing this contribution. We've always wanted someone from the community use the APIs to build and maintain the UI tools which can be of help to everyone in the community.  Super glad to see this coming thru. Can we have a demo of this on 25th Sep? 

@mohitga @vinukumar-vs @amiableanil

**Reply by poojakarma** [reply]
@Krishnaj20 sir 
Thank You !! I'm glad I am able to contribute it to the community.
Yes, we can have demo on 25th September. Please let me know the suitable timing so that We can arrange the meeting.

**Reply by Krishnaj20** [reply]
I've blocked time at 3:30 pm on 25th Sep. Here is the zoom link. Pls share your email ID - will add you to the invite. 

Join Zoom Meeting:
[https://ekstep-org.zoom.us/j/84950133523?pwd=E4GhxfSyDUU7mp5chV53WxxhB9a0bS.1](https://www.google.com/url?q=https://ekstep-org.zoom.us/j/84950133523?pwd%3DE4GhxfSyDUU7mp5chV53WxxhB9a0bS.1&sa=D&source=calendar&ust=1695797058851002&usg=AOvVaw2SMKw5JjZU3OMYnWL0OEAV)

**Reply by poojakarma** [reply]
Ok sir following is my email Id - pooja.karma@tekditechnologies.com

**Reply by Krishnaj20** [reply]
thanks @poojakarma for the demo. Here is the link to the recording:
https://ekstep-org.zoom.us/rec/share/Uk7HSdOZQn5rSCmzh79gyJe5b_FKMiNvEZq5ZFjzew5k1eecdIg2-HRczHG1p3ga.ruLVRhwelI_bTL7p
Passcode: T&rXYN^6

Do keep us posted when you folks integrate the authentication layer so that we can proceed with the next steps regarding the contribution.

**Reply by poojakarma** [reply]
@Krishnaj20  Sir, Thank you for sharing the recording, and I'm glad you found the demo helpful. We'll definitely keep you updated on the progress of integrating the authentication layer.

**Reply by Krishnaj20** [reply]
@poojakarma @baljeetsingh - Any update on the ETA for this contribution please?

**Reply by poojakarma** [reply]
@Krishnaj20 Sir, Minor changes have been implemented that were discussed in meeting. We are now in the process of awaiting approval and merging. We'll update you on ETA and KeyCloak Integration after internal  discussion with team.
@baljeetsingh Sir

**Reply by parthlawate** [reply]
@poojakarma @ManojLondhe can you update on the status of this please ?

**Reply by mohitga** [reply]
@poojakarma can you update the status, let me know when can contribution will be ready for integration with Sunbird Ed. 

cc: @parthlawate
